src/_210413/boj17143/Main_girawhale.java

```diff
@@ -0,0 +1,106 @@
+            상어 한마리 당 움직이는 최대 거리 = 1000 % Math.max(R, C)
+            
+            상어가 이동하고 먹히는 과정은 더하기 연산이므로 제외시켜도 무관
+
+3. 풀이 : 단순 시뮬레이션. 직접 상어를 움직이고 낚시를 해 끝까지 이동하면 되는 문제
+*/
+public class Main_girawhale {
+    static int[] dy = {0, -1, 1, 0, 0};
+    static int[] dx = {0, 0, 0, 1, -1};
+    static int[] dir = {0, 2, 1, 4, 3}; // 뒤집어줄 방향 저장
+
+    public static void main(String[] args) throws IOException {
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        StringTokenizer st = new StringTokenizer(br.readLine());
+        int R = Integer.parseInt(st.nextToken());
+        int C = Integer.parseInt(st.nextToken());
+        int M = Integer.parseInt(st.nextToken());
+
+        Shark[][] map = new Shark[R + 1][C + 1];
+        int[] mod = {0, 2 * R - 2, 2 * C - 2}; // 상하로 움직일때, 좌우로 움직일때 이만큼 이동하면 다시 원상복구된다!
+
+        for (int i = 0; i < M; i++) {
+            Shark shark = new Shark(Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray());
+            shark.speed %= mod[(shark.dir + 1) / 2]; // mod값만큼 이동하면 처음 위치로 돌아오기 때문에 나머지만큼만 이동하도록 설정
+            map[shark.y][shark.x] = shark;
+        }
+
+        int pos = 0; // 낚시왕의 x좌표
+        int ans = 0; // 낚시한 상어무게
+
+        while (pos < C) {
+            pos++;
+
+            // 낚시왕이 위치한 열에서 아래로 내리면서 null이 아니라면 상어가 존재
+            // 위치한 상어의 무게를 답에 더하고 map을 비우자!
+            for (int y = 1; y <= R; y++) {
+                if (map[y][pos] != null) {
+                    ans += map[y][pos].size; // 상어 무게 더해준당
+                    map[y][pos] = null; // map에 상어가 사라졌어요!
+                    break;
+                }
+            }
+
+            Queue<Shark> move = new LinkedList<>(); // 이동한 상어들의 정보들을 담을 Queue
+            for (int i = 1; i <= R; i++) {
+                for (int j = 1; j <= C; j++) {
+                    if (map[i][j] == null) continue; // 상어 없으면 pass
+
+                    Shark shark = map[i][j];
+                    for (int k = 0; k < shark.speed; k++) {
+                        if (shark.dir == 1 && shark.y == 1 || shark.dir == 2 && shark.y == R
+                                || shark.dir == 3 && shark.x == C || shark.dir == 4 && shark.x == 1) {
+                            shark.dir = dir[shark.dir]; // 상어 이동 방향마다 벽에 박으면 뒤집어주기
+                        }
+
+                        shark.y += dy[shark.dir];
+                        shark.x += dx[shark.dir];
+                    }
+
+                    map[i][j] = null; // 상어가 이동했으니까 좌표 비워주고
+                    move.add(shark); // queue에 넣어 한번에 이동하자
+                }
+            }
+
+            while (!move.isEmpty()) {
+                Shark shark = move.poll();
+
+                if (map[shark.y][shark.x] != null) { // map이 null이 아니면 상어가 존재.
+                    if (map[shark.y][shark.x].size < shark.size) // 큰상어 냅두고 작은상어 없애자
+                        map[shark.y][shark.x] = shark;
+                } else map[shark.y][shark.x] = shark; // 현재 칸에 상어 없으니까 그냥 넣기 ㄱ
```

Comment:
쏭쌤! 할 말이 있어여!!! 🤗
여기 중복 코드가 보여요 😗😗 `map[shark.y][shark.x] = shark` 
그리구, 개인적으로 어떤 조건일 때 map에 shark를 저장하는지 한 눈에 안 읽히는 것 같아요 😫

아래와 같은 로직은 어떨까요?? 😚
변수명은 살짝쿵 애매하지만... 😋
```suggestion
		Shark sharkInMap = map[shark.y][shark.x];
		if (sharkInMap == null || (sharkInMap != null && (sharkInMap.size < shark.size))) 
		    map[shark.y][shark.x] = shark;
```